src/index.tsx

```diff
@@ -84,9 +86,19 @@ app.services.promiseCollector.calm().then(() => {
 });
 
 if (window.__PRELOADED_STATE__) {
-  Loadable.preloadReady().then(() => {
-    ReactDOM.hydrate(<app.container />, document.getElementById('root'), doneRendering);
-  });
+  Loadable.preloadReady()
+    .then(() => {
+      // during pre-rendering this happens in src/app/content/hooks/intlHook.ts
+      // it would be nice to consolodate this logic, but in hydration we don't necessarily
+      // want to wait for promiseCollector.calm() before rendering _anything_, so there are some
+      // discrepancies in the flow that make the logic annoying.
```

Comment:
This refers to the `if` branches right? I see you could combine the 2 `then`s in this branch of the `if` by awaiting on createIntl() if needed, but you'd have to make the arrow function async. In the end I don't know if it would be better or worse or matter at all.